SQL function compilation failed when input parameter not referenced

```
> CREATE FUNCTION testing.foo.bar(x integer) RETURNS integer RETURN 1
> SELECT testing.foo.bar(1)
```
Fails with `Compiler failed`.
```
Caused by: java.lang.NullPointerException
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1492)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at com.facebook.presto.sql.relational.SqlFunctionUtils.getSqlFunctionRowExpression(SqlFunctionUtils.java:107)
        at com.facebook.presto.sql.gen.RowExpressionCompiler$Visitor.visitCall(RowExpressionCompiler.java:117)
        at com.facebook.presto.sql.gen.RowExpressionCompiler$Visitor.visitCall(RowExpressionCompiler.java:98)
        at com.facebook.presto.spi.relation.CallExpression.accept(CallExpression.java:107)
        at com.facebook.presto.sql.gen.RowExpressionCompiler.compile(RowExpressionCompiler.java:95)
        at com.facebook.presto.sql.gen.RowExpressionCompiler.compile(RowExpressionCompiler.java:88)
        at com.facebook.presto.sql.gen.PageFunctionCompiler.generateEvaluateMethod(PageFunctionCompiler.java:357)
        at com.facebook.presto.sql.gen.PageFunctionCompiler.definePageProjectWorkClass(PageFunctionCompiler.java:237)
        at com.facebook.presto.sql.gen.PageFunctionCompiler.compileProjectionInternal(PageFunctionCompiler.java:189)
        at com.facebook.presto.sql.gen.PageFunctionCompiler.compileProjection(PageFunctionCompiler.java:163)
        at com.facebook.presto.sql.gen.ExpressionCompiler.lambda$compilePageProcessor$3(ExpressionCompiler.java:108)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.Collections$2.tryAdvance(Collections.java:4727)
        at java.base/java.util.Collections$2.forEachRemaining(Collections.java:4735)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at com.facebook.presto.sql.gen.ExpressionCompiler.compilePageProcessor(ExpressionCompiler.java:109)
        at com.facebook.presto.sql.gen.ExpressionCompiler.compilePageProcessor(ExpressionCompiler.java:96)
        at com.facebook.presto.sql.planner.LocalExecutionPlanner$Visitor.visitScanFilterAndProject(LocalExecutionPlanner.java:1283)
        ... 75 more
```

--- a/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlFunctionUtils.java
+++ b/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlFunctionUtils.java
@@ -103,7 +103,8 @@ public final class SqlFunctionUtils
                 functionMetadata.getArgumentNames().get().stream()
                         .map(Identifier::new)
                         .map(variables::get)
-                        .map(VariableReferenceExpression::getName)
+                        .map(Optional::ofNullable)
+                        .map(variable -> variable.map(VariableReferenceExpression::getName))
                         .collect(toImmutableList()),
                 arguments);
     }
@@ -217,12 +218,14 @@ public final class SqlFunctionUtils
             return ExpressionTreeRewriter.rewriteWith(new ExpressionFunctionVisitor(argumentBindings.build()), function);
         }
 
-        public static RowExpression bindFunctionArguments(RowExpression function, List<String> argumentNames, List<RowExpression> argumentValues)
+        public static RowExpression bindFunctionArguments(RowExpression function, List<Optional<String>> argumentNames, List<RowExpression> argumentValues)
         {
             checkArgument(argumentNames.size() == argumentValues.size(), format("Expect same size for argumentNames (%d) and argumentValues (%d)", argumentNames.size(), argumentValues.size()));
             ImmutableMap.Builder<String, RowExpression> argumentBindings = ImmutableMap.builder();
             for (int i = 0; i < argumentNames.size(); i++) {
-                argumentBindings.put(argumentNames.get(i), argumentValues.get(i));
+                if (argumentNames.get(i).isPresent()) {
+                    argumentBindings.put(argumentNames.get(i).get(), argumentValues.get(i));
+                }
             }
             return RowExpressionTreeRewriter.rewriteWith(new RowExpressionRewriter<Map<String, RowExpression>>()
             {

